apps/omg_db/lib/db.ex

```diff
@@ -31,41 +31,37 @@ defmodule OMG.DB do
   @callback blocks(block_to_fetch :: list()) :: {:ok, list(term)}
   @callback utxos() :: {:ok, list({utxo_pos_db_t, term})}
   @callback utxo(utxo_pos_db_t) :: {:ok, term} | :not_found
-  @callback exit_infos() :: {:ok, list(term)}
-  @callback in_flight_exits_info() :: {:ok, list(term)}
   @callback competitors_info() :: {:ok, list(term)}
-  @callback exit_info({pos_integer, non_neg_integer, non_neg_integer}) :: {:ok, map} | :not_found
   @callback spent_blknum(utxo_pos_db_t()) :: {:ok, pos_integer} | :not_found
   @callback block_hashes(integer()) :: {:ok, list()}
   @callback child_top_block_number() :: {:ok, non_neg_integer()} | :not_found
+  @callback get_single_value(atom()) :: {:ok, term} | :not_found
+  @callback get(atom(), term()) :: {:ok, term} | :not_found
+  @callback get_all_by_type(atom()) :: {:ok, list(term)} | :not_found
 
   # callbacks useful for injecting a specific server implementation
   @callback initiation_multiupdate(GenServer.server()) :: :ok | {:error, any}
   @callback multi_update(term(), GenServer.server()) :: :ok | {:error, any}
   @callback blocks(block_to_fetch :: list(), GenServer.server()) :: {:ok, list()} | {:error, any}
   @callback utxos(GenServer.server()) :: {:ok, list({utxo_pos_db_t, term})} | {:error, any}
   @callback utxo(utxo_pos_db_t, GenServer.server()) :: {:ok, term} | :not_found
-  @callback exit_infos(GenServer.server()) :: {:ok, list(term)} | {:error, any}
-  @callback in_flight_exits_info(GenServer.server()) :: {:ok, list(term)} | {:error, any}
   @callback competitors_info(GenServer.server()) :: {:ok, list(term)} | {:error, any}
-  @callback exit_info({pos_integer, non_neg_integer, non_neg_integer}, GenServer.server()) ::
-              {:ok, map} | :not_found
   @callback spent_blknum(utxo_pos_db_t(), GenServer.server()) :: {:ok, pos_integer} | :not_found
   @callback block_hashes(integer(), GenServer.server()) :: {:ok, list()}
   @callback child_top_block_number(GenServer.server()) :: {:ok, non_neg_integer()} | :not_found
+  @callback get_single_value(atom(), GenServer.server()) :: {:ok, term} | :not_found
+  @callback get(atom(), term(), GenServer.server()) :: {:ok, term} | :not_found
+  @callback get_all_by_type(atom(), GenServer.server()) :: {:ok, list(term)} | :not_found
   @optional_callbacks child_spec: 1,
                       initiation_multiupdate: 1,
                       multi_update: 2,
                       blocks: 2,
                       utxos: 1,
                       utxo: 2,
-                      exit_infos: 1,
-                      in_flight_exits_info: 1,
-                      competitors_info: 1,
-                      exit_info: 2,
                       spent_blknum: 2,
                       block_hashes: 2,
-                      child_top_block_number: 1
+                      child_top_block_number: 1,
+                      get_single_value: 2
```

Comment:
Did not add `get/3` and `get_all_by_type/2` as optional as it is used in the implementation of `PaymentExitInfo`